### PR TITLE
#471 Adds better technique importing for v2 TEF files

### PR DESCRIPTION
**Pull request by ffio1**
Addresses an issue when importing v2 TEF files where techniques, and dynamics were not being imported.

Changes include:
- Dynamics
- Hammer On
- Pull Off
- Natural Harmonic
- Artificial Harmonic
- Palm Mute
- Tapping
- Vibrato
- Tremolo (only 1 duration though. No metadata for duration)
- Bend / Bend + Release
- Let Ring
- Slap
- Ghost Note
- Staccato
- Fade In

### CODE REVIEW

File: desktop/TuxGuitar-tef/src/org/herac/tuxguitar/io/tef/TEInputStream.java [hunk 1]
```diff
@@ -225,13 +225,14 @@ private void readComponents(){
 			
 			if( ((data[2] & 0xff) & 0x1f) > 0  && ((data[2] & 0xff) & 0x1f) <= 25 ){
 				int duration = (data[3] & 0xf);
```
Comment (guiv42) [inline desktop/TuxGuitar-tef/src/org/herac/tuxguitar/io/tef/TEInputStream.java:227]: A problem not directly related to this PR: I was surprised by the encoding of `dynamic` field, as its least significant bit was unused (see `TESongParser.getVelocityFromDynamic`), so I tried to force this bit manually.

It appears this bit corresponds to "double-dotted" attribute. So:
- decoding of `dynamic` is OK, it's the 3 most significant bits of `data[3]`
- but `duration` is encoded on the **5** least significant bits of `data[3]` (and not 4 as currently defined), with most significant bit == "double-dotted"

@ffio1 : do you think you could fix this?

**Review by guiv42 — COMMENTED** [review]
Thank you very much @ffio1 for this PR.
I followed your idea to get an old version of TablEdit for a few tests. I did not test exhaustively your PR, but I just detected an error about decoding of one note's duration (see my remark in the code). Note that this error was there already before this PR.
OK for me to merge this, after 1.6.5 has been released

File: desktop/TuxGuitar-tef/src/org/herac/tuxguitar/io/tef/TEInputStream.java (hunk 1 above)
Comment (ffio1) [reply desktop/TuxGuitar-tef/src/org/herac/tuxguitar/io/tef/TEInputStream.java:227]: @guiv42 Yeah I can fix that!

Comment (ffio1) [reply desktop/TuxGuitar-tef/src/org/herac/tuxguitar/io/tef/TEInputStream.java]: @guiv42 I was able to get my v3 stuff moved to the side temporarily, and committed a fix for this. I did not see the same MSB for double dotted that you were seeing.

What I was seeing that 19, 22, 25, and 28 were the double dotted variants. While I was there, I also fixed the whole note dotted case (31) as it was being imported with the wrong duration.

Comment (guiv42) [reply desktop/TuxGuitar-tef/src/org/herac/tuxguitar/io/tef/TEInputStream.java]: > I did not see the same MSB for double dotted that you were seeing

I did not test as precisely as you did: I was just surprised by this unused bit, tried to set it manually to 1 for one note, and saw a double-dotted note appear in TablEdit. I did not go any further in the analysis.

Thanks!

**Pull request merged**